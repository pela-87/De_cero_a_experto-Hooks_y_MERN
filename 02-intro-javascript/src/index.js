//Funciones en JS
const saludar = function ( nombre ) {
    return `Hola, ${nombre}`;
}



//console.log(saludar('Goku'));

console.log(saludar);