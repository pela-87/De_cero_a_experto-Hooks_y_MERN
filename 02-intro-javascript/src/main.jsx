
//Funciones en JS
const saludar = function ( nombre ) {
    return `Hola, ${nombre}`;
}

//FunciÃ³n flecha
const saludar2 = ( nombre ) => {
    return `Hola, ${nombre}`;
}

const saludar3 = ( nombre ) => `Hola, ${nombre}`;
const saludar4 = () => `Hola Mundo..!!!`;


console.log(saludar('Goku'));
console.log(saludar2('Vegeta'));
console.log(saludar3('Gohan'));
console.log(saludar4());

// const getUser = () => {
//     return {
//         uid: 'ABC123',
//         username: 'Pela87'
//     }
// };

// console.log(getUser());

//Es otra forma de devolver un objeto sin el RETURN.
const getUser = () => (
    {
        uid: 'ABC123',
        username: 'Pela87'
    }
);

console.log(getUser());

