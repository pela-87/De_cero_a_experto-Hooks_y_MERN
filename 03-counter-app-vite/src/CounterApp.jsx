import { useState } from 'react';
//import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {
    
    // function heandleAdd (event) {
        //     console.log(event)
        // }

    //Hooks
    const [counter, setCounter] = useState(value);

    /* Cuando una funcion no tiene mucho que ver con el componente es aconsejable sacarlo a fuera del componente. */
    const heandleAdd = () => { 
        //console.log( event ) 
        // value = 1000;
        // console.log(value);
        
        //Hooks
        //setCounter(counter + 1);
        //setCounter( () => counter + 1);
        setCounter( ( c ) => c + 1);

    }

    const heandleResta = () => setCounter( counter - 1);

    const heandleReset = () => setCounter(value);
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ heandleAdd }> 
                +1 
            </button>
            <button onClick={ heandleResta }> 
                -1 
            </button>
            <button onClick={ heandleReset }> 
                Reset 
            </button>
        </>
    )
};

// CounterApp.prototype = {
//     value:  PropTypes.number
// }
