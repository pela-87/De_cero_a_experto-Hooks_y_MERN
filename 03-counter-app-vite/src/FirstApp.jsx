/*
Se puede mostrar varios elementos dentro de un componte si
se encapsila dentro de un DIV, pero esto genera en la vista
un nuevo DIV que puede romper el doseÃ±o.
*/
/*
export const FirstApp = () => {
    return (
    <div>
        <h1>Primer componente..!!!</h1>
        <p>lalalal </p>
    </div>
    )
};
*/ 

/*
De esta forma se importa un fragmento, para que se pueda enviar
mas de un elemento en el componente y a la vez en el componente
se agrupa dentro de la etiqueta Fragment.
*/
/*
import { Fragment } from "react";

export const FirstApp = () => {
    return (
    <Fragment>
        <h1>Primer componente..!!!</h1>
        <p>lalalal </p>
    </Fragment>
    )
}; 
*/

/*
Esta es una forma mas sencilla de utilizar un Fragment dentro
de la vista y mostrar varios elementos dentro del componente,
que se esta utilizando en la nueva version de React.
*/
/*
export const FirstApp = () => {
    return (
    <>
      <h1>Primer componente..!!!</h1>
      <p>lalalal </p>
    </>
    )
}; 
*/


// const newMessage = 'Pela';
// const newMessage = {
//     Nombre: 'Pela',
//     Edad: 38
// };

// function getNombre (){
//     return newMessage.Nombre + ' ' 
//     + newMessage.Edad;
// };

// export const FirstApp = () => {
//     return (
//     <>
//         {/* <h1> { newMessage.Nombre  } </h1>
//         <h1> { newMessage.Edad  } </h1> */}
//         {/* <h1> { JSON.stringify( newMessage ) } </h1> */}
//         {/* <h1> { getNombre(  ) } </h1> */}
//         <h1>Pela</h1>
//         <p>lalalal </p>
//     </>
//     )
// }; 

/*
//Props

//Pueden venir como props o desestructurado {objeto}.

// export const FirstApp = ( { title, subTitle }) => {
//     //console.log(props);
//     return (
//     <>        
//         <h1>{ title }</h1>
//         <p>Esto es un parrafo.</p>
//         <p>{ subTitle }</p>
//     </>
//     )
// }; 
*/

//PropTypes
import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name }) => {
        
    return (
    <>        
        <h1 data-testid="test-title"> { title } </h1>
        <p>{ subTitle }</p>
        <p>{ subTitle }</p>
        <p>{ name }</p>
    </>
    )
}; 

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Pela',
};