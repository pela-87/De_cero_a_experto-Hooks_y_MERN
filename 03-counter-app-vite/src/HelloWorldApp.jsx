// function App() {
//     return <h1>Hola Mundo..!!</h1>;
// }

export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  )
};

//export const HelloWorldApp = () => <h1>Arrow function</h1>;