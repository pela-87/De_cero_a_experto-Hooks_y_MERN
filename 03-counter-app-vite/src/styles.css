
html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial,  sans-serif;
    font-size: 1.0rem;
    padding: 30px;
}

button {
    padding: 10px;
    font-size: 0.8rem;
    padding-right: 10px;
}