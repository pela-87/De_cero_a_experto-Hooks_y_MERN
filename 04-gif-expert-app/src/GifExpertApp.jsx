import { useState } from 'react';

export const GifExpertApp = () => {
    
    const [ categories, setCategories ] = useState(['One Punch', 'Dragon Ball']);

  return (
    <>
        {/* Titulo */}
        <h1>GifExpertApp</h1>

        {/* Input */}
    <ol>
        { categories.map( (category) => {
            return <li key={ category }>
                        { category }
                   </li>
            })
        }
    </ol>
        {/* Lista de Gif */}
        {/* Gif Item */}
    </>
  )
}

