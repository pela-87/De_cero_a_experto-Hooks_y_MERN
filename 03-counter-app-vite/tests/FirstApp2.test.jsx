import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Prueba en componente <FirstApp> ', () => {
    
    const titulo = 'Hola, soy Goku';
    const subTitulo = 'Soy un subtítulo';

    test ('Debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={ titulo } /> );
        //expect( getByText(titulo) ).toBeTruthy();
        expect( container ).toMatchSnapshot();
    });

    test ('Debe mostrar el mensaje "Hola, Soy Goku"', () => {
        render(<FirstApp title={ titulo } /> );
        //screen.debug();
        expect( screen.getByText(titulo) ).toBeTruthy();
    });

    test ('Debe mostrar el título en un h1', () => {
        render(<FirstApp title={ titulo } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(titulo);
    });

    test('Debe mostrar el subtítulo enviado por props', () => {
        render(<FirstApp title={ titulo } subTitle={ subTitulo } /> );
        expect( screen.getAllByText(subTitulo).length ).toBe(2);
    });
});