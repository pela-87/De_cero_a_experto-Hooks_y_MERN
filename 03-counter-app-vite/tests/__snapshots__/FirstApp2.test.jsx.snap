// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Prueba en componente <FirstApp>  Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Pela
  </p>
</div>
`;
